### PR TITLE
build(packaging): Update nodejs and npm versions

### LINKED ISSUE #394: Update node.js and npm version
Node.js 14 is going to be deprecated. We need to move to the next LTS version and set those versions in package.json.

### PR DESCRIPTION
**Pull request by Devansu-Yadav**
Update Nodejs and NPM to the LTS versions in the Github workflow as well as for build since Nodejs 14 LTS is going to be deprecated.

Output of `Jest` tests using the `npm test` command:

Issue #394 npm test output_1

Output of Cypress test using `npm e2e:static` command:

Issue #394 e2e static Cypress test output_1

Closes #394

### CODE REVIEW

**Review by manekinekko — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -105,7 +105,7 @@
     ]
   },
   "engines": {
-    "node": ">=14.0.0",
+    "node": ">=16.0.0",
```
Comment (sinedied) [inline package.json:108]: It's fine if we target node 16 for our dev/release workflow, but we shouldn't stop support for node 14 right away as a lot of people are still relying on it and it's also a supported runtime for SWA and Functions.

Our test suite runs starting from Node 12+, so as long as we don't have a blocking issue for supporting older Node version I'd suggest that we keep supporting at least all runtimes supported by SWA, ie Node >= 14 IIRC

**Review by sinedied — CHANGES_REQUESTED** [review]

**Comment by Devansu-Yadav** [comment]
@manekinekko @sinedied I hope this PR resolves this issue. Please let me know if any changes are required.

**Review by sinedied — APPROVED** [review]

**Review by manekinekko — APPROVED** [review]

**Comment by manekinekko** [comment]
Thank you @Devansu-Yadav for your great contribution. Your PR is scheduled to be merged as soon as we finish adding the other major features, and merge them to the main branch.

**Comment by sinedied** [comment]
Thanks for this PR @[Devansu-Yadav](https://github.com/Devansu-Yadav)! Would it be possible for you to revert your changes in package-lock.json, as it will cause issues for merging it?

**Comment by Devansu-Yadav** [comment]
> Thanks for this PR @[Devansu-Yadav](https://github.com/Devansu-Yadav)! Would it be possible for you to revert your changes in package-lock.json, as it will cause issues for merging it?

I have reverted the changes in the file 👍

**Pull request merged**